ee_plugin/utils.py

```diff
@@ -375,24 +375,28 @@ def tile_extent(
     ee_image: ee.Image,
     extent: Tuple[float, float, float, float],
     scale: float,
+    projection: str = "EPSG:4326",
 ) -> List[Tuple[float, float, float, float]]:
-    logger.debug(f"Tiling extent {extent} with scale {scale}")
+    logger.debug(f"Tiling extent {extent} with scale {scale}, projection {projection}")
     num_bands = ee_image.bandNames().size().getInfo()
     bytes_per_pixel = num_bands * 2
-    max_bytes = 30 * 1024 * 1024  # Apply safety margin to avoid exceeding EE limit
+    max_bytes = 30 * 1024 * 1024
     max_pixels = max_bytes // bytes_per_pixel
 
     xmin, ymin, xmax, ymax = extent
     width = xmax - xmin
     height = ymax - ymin
 
-    # Calculate tile dimensions in degrees
-    tile_side_meters = math.sqrt(max_pixels) * scale
-    tile_side_degrees = tile_side_meters / 111320
-    tile_width = tile_side_degrees
-    tile_height = tile_side_degrees
+    tile_side = math.sqrt(max_pixels) * scale
+
+    # If using geographic coordinates, convert tile side from meters to degrees
+    if projection == "EPSG:4326":
+        tile_width = tile_side / 111320
+        tile_height = tile_width
+    else:
+        tile_width = tile_side
+        tile_height = tile_side
```

Comment:
```suggestion
    tile_width = tile_height = tile_side / 111320 if projection == "EPSG:4326" else tile_side
```

This is maybe a simpler way to write this?

Or maybe:

```suggestion
    if projection == "EPSG:4326":
        tile_side = tile_side / 111320
    tile_width = tile_height = tile_side
```